icu4c/source/i18n/number_rounding.cpp

```diff
@@ -283,6 +283,10 @@ FractionPrecision Precision::constructFraction(int32_t minFrac, int32_t maxFrac)
     settings.fMaxFrac = static_cast<digits_t>(maxFrac);
     settings.fMinSig = -1;
     settings.fMaxSig = -1;
+    // The following are not used under RND_FRACTION but initialize to avoid
+    // warning.
+    settings.fPriority = UNUM_ROUNDING_PRIORITY_RELAXED;
+    settings.fRetain = false;
```

Comment:
Instead of doing this, another alternative which requires less code and will continue to work even if new fields are added in the future would be to just default-initialize `settings` before setting only those values that need to be set to something other than the default:

```
    FractionSignificantSettings settings{};
```